Document the buildVirtualDevice task (#655)
MicroEJ/docs: SDK6UserGuide/appendix/virtualDevice.rst

.. _virtual_device_chapter:

Virtual Device
==============

This chapter describes the structure of a :ref:`section.virtual.device`.

.. _virtual_device_structure:

Structure
---------

A Virtual Device is structured as follows:

.. code-block::

   |- virtualDevice
   |   |- installed-applications
   |   |- javaLibs
   |   |- MICROJVM
   |   |   |- javaLibs 
   |   |   |- soar    
   |   |- main-application
   |   |- mocks
   |   |- options
   |   |   |- application.properties 
   |   |- resources
   |   |- s3
   |   |- scripts
   |   |   |- init-vd 
   |   |   |   |- vd-init.xml 
   |   |- tools
   |   |- architecture.properties
   |   |- release.properties
   |   |- veePort.properties
   |   |- workbenchExtension*.jar

> The Virtual Device contains the Simulation part files of the VEE Port used to build it: 

Reviewer: as for JavaDoc, only capitalize the first letter and end with a dot if the item is a proper sentence. See https://www.grammarly.com/blog/bullet-points/

Author: I fixed the bulleted lists.

> - the ``javaLibs/`` folder, that contains the Generic Foundation Libraries.

Reviewer: ```suggestion
- the ``javaLibs/`` folder, that contains the Foundation Libraries which are common to MICROJVM and S3
```

- the ``linker/`` folder, that contains the Linker jar files. This folder is not embedded in the Virtual Device if an Architecture ``8.0.0`` is used.  

> - the ``MICROJVM/javaLibs/`` folder, that contains the Foundation Libraries specific to the MicroEJ Core Engine.

Reviewer: I think that there should only be one entry for MICROJVM (as for S3). You can explain that it only contains files from VEE Port that are needed for building .fo files

Author: Done.

- the ``MICROJVM/soar/`` folder, that contains the Jar files of the SOAR.

- the ``mocks/`` folder, that contains the Jar files of the mocks for Foundation Libraries.

- the ``resources/`` folder, that contains the OS specific libraries.

> - the ``s3/`` folder. 

Reviewer: ```suggestion
- the ``S3/`` folder, that contains the Simulator, HIL and Foundation Libraries specific to the Simulator 
```

- the ``scripts/`` folder.

- the ``tools/`` folder.

- the ``workbenchExtension*.jar`` files.

The following elements are also embedded in the Virtual Device:

> - a Kernel Application, whose WPK file is extracted in the ``main-application`` folder of the Virtual Device.

Reviewer: ```suggestion
- a Kernel Application, whose WPK file is extracted in the ``main-application`` folder of the Virtual Device
```

- the WPK files of System Applications that are extracted in their own folder, in the ``installed-applications/`` folder that is empty by default.

.. note::

   System Applications can only be embedded in a Multi-Sandboxed Virtual Device.
   In case of a Mono-Sandboxed Virtual Device, the ``installed-applications/`` folder is always empty.

- The ``options/application.properties`` that contains the Application properties and the properties of the VEE Port used to build the Virtual Device.

> - The ``scripts/init-vd/vd-init.xml`` script that allows to enable or not the Virtual Device. If the Virtual Device is not enabled, the Application main class specified by the user is launched on the VEE Port. 

Reviewer: ```suggestion
- the ``scripts/init-vd/vd-init.xml`` script, that allows to enable or not the Virtual Device. If the Virtual Device is not enabled, the Application main class specified by the user is launched on the VEE Port
```
